frontend/src/pages/index.js

```diff
@@ -40,6 +42,30 @@ const ProjectCard = ({ name, full_name, description, owner }) => {
   )
 }
 
+export const getStaticProps = async () => {
+  const res = await fetch(`${giteaApiUrl}/repos/search`, {
+    method: 'GET',
+    mode: 'cors',
+    sort: 'updated',
+    order: 'desc',
+    q: undefined,
+  })
+
+  if (res.ok) {
+    return {
+      props: {
+        repos: res.data,
+      },
+    }
+  } else {
+    return {
+      props: {
+        repos: [],
+      },
+    }
+  }
+}
+
```

Comment:
I know there's a function in the gitea client doing the same thing though I preferred to make the commit in a single file